packages/docusaurus-theme-slashid/src/domain.test.ts

```diff
@@ -0,0 +1,61 @@
+/* ============================================================================
+ * Copyright (c) SlashID
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ * ========================================================================== */
+
+import { shouldPathRender } from "./domain";
+
+const TEST_USER = { getGroups: () => [] };
+
+describe("domain", () => {
+  describe("shouldPathRender", () => {
+    test("should return true if no privatePaths", () => {
+      expect(shouldPathRender("/foo", undefined, undefined)).toBe(true);
+    });
+
+    test("should return true if no path", () => {
+      expect(shouldPathRender(undefined, [{ path: "/foo" }], undefined)).toBe(
+        true
+      );
+    });
+
+    test("should return false if no user and the path matches", () => {
+      expect(shouldPathRender("/foo", [{ path: "/foo" }], undefined)).toBe(
+        false
+      );
+    });
+
+    test("should return true if a user is present and no groups are required", () => {
+      // @ts-expect-error cannot import User for some reason
+      expect(shouldPathRender("/foo", [{ path: "/foo" }], TEST_USER)).toBe(
+        true
+      );
+    });
+
+    test("should return true if a user is present and the user has the required groups", () => {
+      expect(
+        // @ts-expect-error cannot import User for some reason
+        shouldPathRender("/foo", [{ path: "/foo", groups: ["bar"] }], {
+          getGroups: () => ["bar"],
+        })
+      ).toBe(true);
+    });
+
+    test("should return false if a user is present and the user does not have the required groups", () => {
+      expect(
+        // @ts-expect-error cannot import User for some reason
+        shouldPathRender("/foo", [{ path: "/foo", groups: ["bar"] }], {
+          getGroups: () => [],
+        })
+      ).toBe(false);
+    });
+
+    test("should return false if a user is not present and the provided regex matches", () => {
+      expect(
+        shouldPathRender("/foo/bar", [{ path: /^\/foo/ }], undefined)
```

Comment:
Should we support globs?

We could use something like [micromatch](https://www.npmjs.com/package/micromatch) to do it cheaply

Reply:
Ahh I started from globs but then when researching libraries I couldn't find anything really small - even micromatch adds 35kb minified. Globs would surely look better though!

Do you think it's worth the cost? I tested it with a couple of simple regex and it worked OK, but globs would be more intuitive.

Reply:
I'm biased but I would say it's worth it, regex is scary

Reply:
Maybe a glob-to-regex library is smaller in size than a standalone matcher? E.g., https://www.npmjs.com/package/glob-to-regexp

Reply:
@paulo-raca this is interesting, it's under 1kB. Let me test it out!

Reply:
Done!